src/lib/logstore/log_store.cpp

```diff
@@ -283,13 +283,17 @@ void HomeLogStore::flush(logstore_seq_num_t upto_lsn) {
 }
 
 bool HomeLogStore::rollback(logstore_seq_num_t to_lsn) {
-    // Validate if the lsn to which it is rolledback to is not truncated.
-    auto ret = m_records.status(to_lsn + 1);
-    if (ret.is_out_of_range) {
-        HS_LOG_ASSERT(false, "Attempted to rollback to {} which is already truncated", to_lsn);
+    //Fast path
+    if (to_lsn == m_tail_lsn.load()) {
+	return true;
+    }
+
+    if (to_lsn > m_tail_lsn.load() || to_lsn < m_start_lsn.load()) {
+        HS_LOG_ASSERT(false, "Attempted to rollback to {} which is not in the range of [{}, {}]", to_lsn, m_start_lsn.load(), m_tail_lsn.load());
         return false;
     }
 
```

Comment:
Can we also have an assert here for `HS_DEBUG_ASSERT(m_records.status(to_lsn + 1).is_out_of_range == false)`?

Reply:
there are two angle, 

- from api point of view, why we want to assert  for a no-op rollback?  i. e  if last log is 100, rollback to 100 is a valid request
- from the implementation of streamtracker.   the is_out_of_range will not be set for LSN > upper_bound.  
 https://github.com/eBay/sisl/blob/ed773dbb0c7e53f46b29374e477f4be3dc1db99b/include/sisl/fds/stream_tracker.hpp#L128